### PR TITLE
Add more references - Testing Business Logic Data Validation

### PR DESCRIPTION
**Pull request by RiieCco**
This PR closes OWASP/wstg#354

- [x] This PR handles the issue and requires no additional PRs.
- [x] You have validated the need for this change.

**What did this PR accomplish?**

- Added new references to the "References" section
- Added an extra example

Thank you for your contribution!

### CODE REVIEW

File: document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md [hunk 1]
```diff
@@ -58,6 +66,10 @@ All [Input Validation](../07-Input_Validation_Testing/README.md) test cases.
 
 [Beginning Microsoft Visual Studio LightSwitch Development](https://books.google.com/books?id=x76L_kaTgdEC&pg=PA280&lpg=PA280&dq=business+logic+example+valid+data+example&source=bl&ots=GOfQ-7f4Hu&sig=4jOejZVligZOrvjBFRAT4-jy8DI&hl=en&sa=X&ei=mydYUt6qEOX54APu7IDgCQ&ved=0CFIQ6AEwBDgK#v=onepage&q=business%20logic%20example%20valid%20data%20example&f=false)
 
+[OWASP Proactive controls(C5) - Validate all inputs](https://owasp.org/www-project-proactive-controls/v3/en/c5-validate-inputs)
+
+[OWASP Cheatsheet series - Input_Validation_Cheat_Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html)
+
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md]:
```suggestion
[OWASP Proactive Controls (C5) - Validate All Inputs](https://owasp.org/www-project-proactive-controls/v3/en/c5-validate-inputs)

[OWASP Cheatsheet Series - Input_Validation_Cheat_Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Input_Validation_Cheat_Sheet.html)

```

File: document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md [hunk 2]
```diff
@@ -24,6 +24,14 @@ Suppose you manage a multi-tiered e-commerce site that allows users to order car
 
 Many credit card systems are now downloading account balances nightly so the customers can check out more quickly for amounts under a certain value. The inverse is also true. If I pay my credit card off in the morning I may not be able to use the available credit in the evening. Another example may be if I use my credit card at multiple locations very quickly it may be possible to exceed my limit if the systems are basing decisions on last night’s data.
 
+### Example 3
+
+**Distributed Denial of Dollar (DDo$):**
+This was a campaign that was proposed by the founder of the website " The Pirate Bay " against the law firm who brought prosecutions against them. The goal was to take advantage of errors in the design of business features and in the process of credit transfer validation.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md]:
```suggestion
This was a campaign that was proposed by the founder of the website "The Pirate Bay" against the law firm who brought prosecutions against "The Pirate Bay". The goal was to take advantage of errors in the design of business features and in the process of credit transfer validation.
```

File: document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md [hunk 3]
```diff
@@ -24,6 +24,14 @@ Suppose you manage a multi-tiered e-commerce site that allows users to order car
...
+
+This attack was perfomed by sending very small amounts of money of 1 SEK ($0.13) to the law firm.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md]:
```suggestion
This attack was perfomed by sending very small amounts of money of 1 SEK ($0.13 USD) to the law firm.
```

Assuming US dollars.

File: document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md [hunk 4]
```diff
@@ -24,6 +24,14 @@ Suppose you manage a multi-tiered e-commerce site that allows users to order car
...
+The bank account to which the payments were directed had only 1000 free transfers, after which any transfers have a surcharge  for the account holder (2 SEK). After the first thousand internet transactions every 1 SEK donation to the law firm will actually end up costing it 1 SEK instead.
```
Comment (kingthorin) [inline document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md]:
```suggestion
The bank account to which the payments were directed had only 1000 free transfers, after which any transfers have a surcharge for the account holder (2 SEK). After the first thousand internet transactions every 1 SEK donation to the law firm will actually end up costing it 1 SEK instead.
```

File: document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md [hunk 5]
```diff
@@ -24,6 +24,14 @@ Suppose you manage a multi-tiered e-commerce site that allows users to order car
 
 Many credit card systems are now downloading account balances nightly so the customers can check out more quickly for amounts under a certain value. The inverse is also true. If I pay my credit card off in the morning I may not be able to use the available credit in the evening. Another example may be if I use my credit card at multiple locations very quickly it may be possible to exceed my limit if the systems are basing decisions on last night’s data.
 
+### Example 3
+
+**Distributed Denial of Dollar (DDo$):**
```
Comment (ThunderSon) [inline document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md]:
```suggestion
**[Distributed Denial of Dollar (DDo$)](https://news.hitb.org/content/pirate-bay-proposes-distributed-denial-dollars-attack-ddo):**
```

Comment (kingthorin) [reply document/4-Web_Application_Security_Testing/10-Business_Logic_Testing/01-Test_Business_Logic_Data_Validation.md]: I'll knock together a PR with that.

**Pull request merged**